internal/services/delegation.go

```diff
@@ -56,6 +60,41 @@ func (s *Service) processNewBTCDelegationEvent(
 		)
 	}
 
+	stakingTxHash, parseErr := chainhash.NewHashFromStr(delegationDoc.StakingTxHashHex)
+	if parseErr != nil {
+		return types.NewError(
+			http.StatusInternalServerError,
+			types.InternalServiceError,
+			fmt.Errorf("failed to parse staking tx hash: %w", parseErr),
+		)
+	}
+
+	stakingOutputPkScript, parseErr := hex.DecodeString(delegationDoc.StakingOutputPkScript)
+	if parseErr != nil {
+		return types.NewError(
+			http.StatusInternalServerError,
+			types.InternalServiceError,
+			fmt.Errorf("failed to decode staking output pk script: %w", parseErr),
+		)
+	}
```

Comment:
I would add a function to parse the the events into something like a `parsedBTCDelegationCreatedEvent` struct (to replace the validate function)